mad-dog/src/tests/test_ursmBlockSaturation.js

```diff
@@ -0,0 +1,142 @@
+}
+
+const maxIndexingTimeout = 15000
+
+const waitForBlock = async (libs, targetBlockNumber) => {
+  let latestIndexedBlock = await getLatestIndexedBlock(libs.getDiscoveryNodeEndpoint())
+  const startTime = Date.now()
+  while (Date.now() - startTime < maxIndexingTimeout) {
+    latestIndexedBlock = await getLatestIndexedBlock(libs.getDiscoveryNodeEndpoint())
+    if (latestIndexedBlock >= targetBlockNumber) {
+      logger.info(`Discovery Node has indexed block #${targetBlockNumber}!`)
+      return true
+    }
+  }
+  throw new Error(`Failed to reach ${targetBlockNumber} in ${maxIndexingTimeout}`)
+}
+
+// Insert 2 records for a given user in a single block
+// 3 Transactions total: 1 replica set update, 1 bio update, 1 name update
+// UserFactory will reduce both txs into a single record
+// UserRelicaSetManager will insert another record
+// Calling this function 2x for a single user should result in 2 is_current=False records
+//    for a single block on the second update
+const submitParallelUserTxs = async(executeAll) => {
+  await executeAll(async (libs, i) => {
+    // Retrieve user id if known from walletIndexToUserIdMap
+    const userId = walletIndexToUserIdMap[i]
+    let sameBlockTxSubmitted = false
+    // Array of block numbers
+    let blockResponses = []
+    while(!sameBlockTxSubmitted) {
+        let usrReplicaInfoFromContract = await libs.getUserReplicaSet(userId)
+        logger.info(`userId: ${userId}: submitParallelUserTxs`)
+        let primary = usrReplicaInfoFromContract.primaryId
+        let secondaries = usrReplicaInfoFromContract.secondaryIds
+        let newPrimary = parseInt(secondaries[0])
+        let newSecondaries = [primary, secondaries[1]].map(x=>parseInt(x))
+        logger.info(`userId: ${userId} | P: ${primary}->${newPrimary}`)
+        logger.info(`userId: ${userId} | S1: ${secondaries[0]}->${newSecondaries[0]}`)
+        logger.info(`userId: ${userId} | S2: ${secondaries[1]}->${newSecondaries[1]}`)
+
+        let randomUserInfo = genRandomUsers(1)[0]
+        // Issue 3 concurrent update operations
+        blockResponses = await Promise.all([
+            (async () => {
+                let tx = await libs.updateReplicaSet(userId, newPrimary, newSecondaries)
+                console.log(`userId: ${userId} | Returning ${tx.blockNumber}`)
+                return tx.blockNumber
+            })(),
+            (async () => {
+                let bioTx = await libs.updateBio(userId, randomUserInfo.bio)
+                console.log(`userId: ${userId} | Returning bioTx: ${bioTx.txReceipt.blockNumber}`)
+                return bioTx.txReceipt.blockNumber
+            })(),
+            (async() => {
+                let nameTx = await libs.updateBio(userId, randomUserInfo.name)
+                console.log(`userId: ${userId} | Returning nameTx: ${nameTx.txReceipt.blockNumber}`)
+                return nameTx.txReceipt.blockNumber
+            })()
+        ])
+        sameBlockTxSubmitted = (
```

Comment:
iiuc, the way this is ensured is by the `-b` flag in ganache land, right? since that's not enforced in code, maybe we can make a comment here to explain to the future why we know this happens?

Reply:
good point, that is the only reason why this is possible